### PR TITLE
Attempt to boost sections for certain searches

### LINKED ISSUE #602: Improve internal search results
Review the common words that people use in our internal search bar, and see if the terms bring up the correct pages. If not, determine whether the weight of the keywords can be adjusted in Read The Docs per page.

### PR DESCRIPTION
**Pull request by begriffs**
Fixes #602 

One challenge in testing this is I think the search results will differ on our pull request preview app and actual RTD. According to their [docs](https://read-the-docs.readthedocs.io/en/latest/features.html#search) they run a Haystack and Solr instance for full-text searching.

If the changes in this PR look OK we can merge to master, and try it out on `latest` to see how well it performs. If good we can cherry pick to 7.2, else keep adjusting.

The changes in here based on the email ideas from @samay-sharma

### CODE REVIEW

File: dist_tables/extensions.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
+Citus provides distributed functionality by extending PostgreSQL using the hook and extension APIs. This allows users to benefit from the features that come with the rich PostgreSQL ecosystem. These features include, but aren’t limited to, support for a wide range of `data types <http://www.postgresql.org/docs/current/static/datatype.html>`_ (including semi-structured data types like jsonb and hstore), `operators and functions <http://www.postgresql.org/docs/current/static/functions.html>`_, full text search, and other extensions such as `PostGIS <http://postgis.net/>`_ and `HyperLogLog <https://github.com/aggregateknowledge/postgresql-hll>`_. Further, proper use of the extension APIs enable compatibility with standard PostgreSQL tools such as `pgAdmin <http://www.pgadmin.org/>`_ and `pg_upgrade <http://www.postgresql.org/docs/current/static/pgupgrade.html>`_.
```
Comment (sumedhpathak) [inline dist_tables/extensions.rst]: Whats changed here?

File: cloud/availability.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .. _ha:
 
-Availability and Recovery
-#########################
+Backup, Availability, and Replication
```
Comment (sumedhpathak) [inline cloud/availability.rst]: Any reason to change Recovery to Replication?

Comment (begriffs) [reply cloud/availability.rst]: Just to catch that search keyword, which is a common one.

File: dist_tables/extensions.rst (hunk 1 above)
Comment (begriffs) [reply dist_tables/extensions.rst]: pg_backup was removed, because it was matching searches for "backup"

**Pull request merged**